x-pack/plugins/ml/server/lib/capabilities/check_capabilities.ts

```diff
@@ -22,12 +22,12 @@ import {
 } from './errors';
 
 export function capabilitiesProvider(
-  mlClusterClient: ILegacyScopedClusterClient,
+  client: IScopedClusterClient,
```

Comment:
maybe we should create our own interface for this one to swap in the future if needed?